pkg/planner/core/logical_join.go

```diff
@@ -300,13 +300,11 @@ func (p *LogicalJoin) PruneColumns(parentUsedCols []*expression.Column, opt *opt
 	if err != nil {
 		return nil, err
 	}
-	addConstOneForEmptyProjection(p.Children()[0])
```

Comment:
Now that we remove `addConstOneForEmptyProjection` here, it probably means we can also remove that in `(*LogicalApply).PruneColumns()`. We can have a check.

Reply:
Yes, i forgot to remove it. Last week is too busy(